junit5/src/main/java/com/linecorp/armeria/testing/server/ServiceRequestContextCaptor.java

```diff
@@ -0,0 +1,169 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.testing.server;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+import com.linecorp.armeria.server.HttpService;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.SimpleDecoratingHttpService;
+
+/**
+ * Captures the {@code ServiceRequestContext}s.
+ *
+ * <p>Example:
+ * <pre>{@code
+ * class ServiceRequestContextCaptorTest {
+ *     @RegisterExtension
+ *     static final ServerExtension server = new ServerExtension() {
+ *         @Override
+ *         protected void configure(ServerBuilder sb) throws Exception {
+ *             sb.service("/hello", (ctx, req) -> HttpResponse.of(200));
+ *         }
+ *     };
+ *     @Test
+ *     void test() {
+ *         final ServiceRequestContextCaptor captor = server.requestContextCaptor();
+ *         client.get("/hello").aggregate().join();
+ *         assertThat(captor.size()).isEqualTo(1);
+ *     }
+ * }
+ * }</pre>
+ *
+ * <p>Example: use {@code ServiceRequestContextCaptor} manually
+ * <pre>{@code
+ * class ServiceRequestContextCaptorTest {
+ *     static final ServiceRequestContextCaptor captor = new ServiceRequestContextCaptor();
+ *     static final Server server = Server.builder()
+ *                                        .decorator(captor.decorator())
+ *                                        .service("/hello", (ctx, req) -> HttpResponse.of(200))
+ *                                        .build();
+ *
+ *     @BeforeAll
+ *     static void beforeClass() {
+ *         server.start().join();
+ *     }
+ *
+ *     @AfterAll
+ *     static void afterClass() {
+ *         server.stop().join();
+ *     }
+ *
+ *     @Test
+ *     void test() {
+ *         final WebClient client = WebClient.builder("http://127.0.0.1:" + server.activeLocalPort()).build();
+ *         client.get("/hello").aggregate().join();
+ *         assertThat(captor.size()).isEqualTo(1);
+ *     }
+ * }
+ * }</pre>
+ */
+@UnstableApi
+public final class ServiceRequestContextCaptor {
+    private static final int DEFAULT_TIMEOUT_IN_SECONDS = 15;
+
+    private final BlockingQueue<ServiceRequestContext> serviceContexts =
+            new LinkedBlockingDeque<>();
+
+    /**
+     * Creates a new decorator to capture the {@link ServiceRequestContext}s.
+     */
+    public Function<? super HttpService, ? extends HttpService> decorator() {
```

Comment:
For consistency like others:
https://github.com/line/armeria/blob/52675f74413e0e772bd731376ac970d2d2f6c8a6/core/src/main/java/com/linecorp/armeria/server/logging/LoggingService.java#L59-L61
```suggestion
    public Function<? super HttpService, ? extends HttpService> newDecorator() {
```